experimental/framework/tree-react-api/src/testExports.ts

```diff
@@ -0,0 +1,104 @@
+and its tests are used to test importing them.
+See ./test/importer.spec.ts for for the imports.
+
+This could be part of the tree package and its tests,
+however that is less representative of actual use, and thus fails to detect some issues which impact real users (like `error TS2742` see note  below).
+*/
+
+// This file is full of test exports to ensure the types using tree types can be exported.
+// Requiring these to have docs seems unnecessary.
+/* eslint-disable jsdoc/require-jsdoc */
+
+import {
+	SchemaFactory,
+	TreeViewConfiguration,
+	type NodeFromSchema,
+	type ValidateRecursiveSchema,
+	// To diagnose `error TS2742: The inferred type of ...` errors,
+	// enable this import then look for use of it in the generated d.ts file fo find what needs to be moved out of InternalTypes
+	// // eslint-disable-next-line unused-imports/no-unused-imports
+	// InternalTypes,
+} from "@fluidframework/tree";
+
+// Due to limitation of the TypeScript compiler, errors like the following can be produced when exporting types from another package:
+// error TS2742: The inferred type of 'Inventory' cannot be named without a reference to '../node_modules/@fluidframework/tree/lib/internalTypes.js'. This is likely not portable. A type annotation is necessary.
+// Tree needs tests to make sure it does not trigger such errors, but they can't easily be done from within the tree package.
+// Thus tests for this case are included here, in a package which uses tree.
+
+export const schema = new SchemaFactory("com.example");
+
+export const leafAlias = schema.number;
+
+export class Point extends schema.object("Point", {
+	x: schema.number,
+}) {}
+
+export class Note extends schema.object("Note", {
+	text: schema.string,
+	location: schema.optional(Point),
+}) {}
+
+export class NodeMap extends schema.map("NoteMap", Note) {}
+export class NodeList extends schema.array("NoteList", Note) {}
+
+export class Canvas extends schema.object("Canvas", { stuff: [NodeMap, NodeList] }) {}
+
+export const POJO = schema.object("POJO", { stuff: [NodeMap, NodeList] });
+export type POJO = NodeFromSchema<typeof POJO>;
+
+export const config = new TreeViewConfiguration({ schema: Canvas });
+
+// Recursive cases
+// This lint rule doesn't work well with our schema when using the lazy format
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+export class RecursiveObject extends schema.objectRecursive("RO", {
+	x: [() => RecursiveObject, schema.number],
+}) {}
+{
+	type _check = ValidateRecursiveSchema<typeof RecursiveObject>;
+}
+
+export const recursiveField = schema.optionalRecursive([() => RecursiveObject, schema.number]);
+
+export class RecursiveMap extends schema.mapRecursive("RM", [() => RecursiveMap]) {}
+{
+	type _check = ValidateRecursiveSchema<typeof RecursiveMap>;
+}
+
+// TODO: export once ` error TS2310: Type 'RecursiveArray' recursively references itself as a base type.` in the d.ts file is fixed.
```

Comment:
Similar issue to the map one being addressed here? Just planned for a subsequent PR, presumably?

Reply:
I'll have a follow-up PR with a workaround for that case. Sadly the same approach didn't fix it so I did something less nice.